src/averageAges.js

```diff
@@ -55,7 +73,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenWithChil = people
+    .filter(child => child.mother && people
+      .find(person => person.name === child.mother));
+
+  const withSonOnly = womenWithChil
+    .filter(person => person.sex === 'm');
+
+  const ageDifference = array => array.reduce((sum, child) =>
+    sum + child.born - people
+      .find(person => person.name === child.mother).born, 0) / array.length;
+
+  return onlyWithSon
+    ? ageDifference(withSonOnly) : ageDifference(womenWithChil);
```

Comment:
Looks more readable this way
```suggestion
  return onlyWithSon
    ? ageDifference(withSonOnly) 
    : ageDifference(womenWithChil);
```